### ISSUE TITLE
Track is Not Playable, Tidal Login OK

### ISSUE THREAD

**Issue by n10b** [tehkillerbee/mopidy-tidal#21]
Maybe there's problem with Tidal API or token...

```
pi@raspberrypi:~ $ sudo journalctl -u mopidy
May 20 11:32:46 raspberrypi systemd[1]: Starting Mopidy music server...
May 20 11:32:46 raspberrypi systemd[1]: Started Mopidy music server.
May 20 11:32:52 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:52,108 [3163:MainThread] mopidy.__main__
May 20 11:32:52 raspberrypi mopidy[3163]:   Starting Mopidy 3.0.2
May 20 11:32:52 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:52,622 [3163:MainThread] mopidy.config
May 20 11:32:52 raspberrypi mopidy[3163]:   Loading config from builtin defaults
May 20 11:32:52 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:52,767 [3163:MainThread] mopidy.config
May 20 11:32:52 raspberrypi mopidy[3163]:   Loading config from file:///etc/mopidy/mopidy.conf
May 20 11:32:52 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:52,823 [3163:MainThread] mopidy.config
May 20 11:32:52 raspberrypi mopidy[3163]:   Loading config from command line options
May 20 11:32:59 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:59,292 [3163:MainThread] mopidy.__main__
May 20 11:32:59 raspberrypi mopidy[3163]:   Enabled extensions: musicbox_webclient, youtube, stream, softwaremixer, mpd, iris, tidal, alsamixer, http, file, m3u
May 20 11:32:59 raspberrypi mopidy[3163]: INFO     2020-05-20 11:32:59,299 [3163:MainThread] mopidy.__main__
May 20 11:32:59 raspberrypi mopidy[3163]:   Disabled extensions: none
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,373 [3163:MainThread] mopidy.commands
May 20 11:33:08 raspberrypi mopidy[3163]:   Starting Mopidy mixer: SoftwareMixer
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,403 [3163:MainThread] mopidy.commands
May 20 11:33:08 raspberrypi mopidy[3163]:   Starting Mopidy audio
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,442 [3163:MainThread] mopidy.commands
May 20 11:33:08 raspberrypi mopidy[3163]:   Starting Mopidy backends: FileBackend, M3UBackend, StreamBackend, YouTubeBackend, TidalBackend
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,698 [3163:Audio-2] mopidy.audio.actor
May 20 11:33:08 raspberrypi mopidy[3163]:   Audio output set to "alsasink"
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,887 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:33:08 raspberrypi mopidy[3163]:   using bs4API
May 20 11:33:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:08,962 [3163:TidalBackend-7] mopidy_tidal.backend
May 20 11:33:08 raspberrypi mopidy[3163]:   Connecting to TIDAL.. Quality = LOSSLESS
May 20 11:33:10 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:10,760 [3163:TidalBackend-7] mopidy_tidal.backend
May 20 11:33:10 raspberrypi mopidy[3163]:   TIDAL Login OK
May 20 11:33:10 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:10,771 [3163:MainThread] mopidy.commands
May 20 11:33:10 raspberrypi mopidy[3163]:   Starting Mopidy core
May 20 11:33:10 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:10,952 [3163:MainThread] mopidy.commands
May 20 11:33:10 raspberrypi mopidy[3163]:   Starting Mopidy frontends: HttpFrontend, YouTubeAutoplayer, MpdFrontend, IrisFrontend
May 20 11:33:10 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:10,967 [3163:HttpFrontend-10] mopidy.http.actor
May 20 11:33:10 raspberrypi mopidy[3163]:   HTTP server running at [::ffff:0.0.0.0]:6680
May 20 11:33:11 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:11,002 [3163:MainThread] mopidy_mpd.actor
May 20 11:33:11 raspberrypi mopidy[3163]:   MPD server running at [::ffff:0.0.0.0]:6600
May 20 11:33:11 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:11,092 [3163:IrisFrontend-13] mopidy_iris.core
May 20 11:33:11 raspberrypi mopidy[3163]:   Starting Iris 3.48.0
May 20 11:33:11 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:11,099 [3163:MainThread] mopidy.commands
May 20 11:33:11 raspberrypi mopidy[3163]:   Starting GLib mainloop
May 20 11:33:11 raspberrypi mopidy[3163]: INFO     2020-05-20 11:33:11,126 [3163:HttpServer] mopidy.internal.path
May 20 11:33:11 raspberrypi mopidy[3163]:   Creating dir file:///var/lib/mopidy/.local/share/mopidy/http
May 20 11:38:30 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:30,775 [3163:TidalBackend-7] mopidy_tidal.search
May 20 11:38:30 raspberrypi mopidy[3163]:   Searching Tidal for:  {'any': ['didi kempot keroncong']}
May 20 11:38:33 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:33,102 [3163:TidalBackend-7] mopidy_tidal.search
May 20 11:38:33 raspberrypi mopidy[3163]:   Searching Tidal for:  {'album': ['didi kempot keroncong']}
May 20 11:38:39 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:39,429 [3163:TidalBackend-7] mopidy_tidal.search
May 20 11:38:39 raspberrypi mopidy[3163]:   Searching Tidal for:  {'artist': ['didi kempot keroncong']}
May 20 11:38:41 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:41,335 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:38:41 raspberrypi mopidy[3163]:   youtube LibraryProvider.search "{'any': ['didi kempot keroncong']}"
May 20 11:38:41 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:41,342 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:38:41 raspberrypi mopidy[3163]:   Searching YouTube for query "didi kempot keroncong"
May 20 11:38:41 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:41,344 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:38:41 raspberrypi mopidy[3163]:   session.get triggered: search
May 20 11:38:46 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:46,608 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:38:46 raspberrypi mopidy[3163]:   youtube LibraryProvider.search "{'album': ['didi kempot keroncong']}"
May 20 11:38:46 raspberrypi mopidy[3163]: INFO     2020-05-20 11:38:46,642 [3163:YouTubeBackend-6] mopidy_youtube
May 20 11:38:46 raspberrypi mopidy[3163]:   youtube LibraryProvider.search "{'artist': ['didi kempot keroncong']}"
May 20 11:39:06 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:06,333 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:06 raspberrypi mopidy[3163]:   Lookup uris 'tidal:album:85081120'
May 20 11:39:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:08,064 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:08 raspberrypi mopidy[3163]:   Returning 12 tracks
May 20 11:39:08 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:08,314 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:08 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081121'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,747 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,757 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081122'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,768 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,773 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081123'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,782 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,787 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081124'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,796 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,800 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081125'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,810 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,815 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081126'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,824 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,829 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081127'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,838 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,842 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081128'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,851 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,856 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081129'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,866 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,870 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081130'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,879 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,884 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081131'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,893 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,898 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081132'
May 20 11:39:09 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:09,907 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:09 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:17 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:17,127 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:17 raspberrypi mopidy[3163]:   Lookup uris 'tidal:track:4889383:85081120:85081126'
May 20 11:39:17 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:17,140 [3163:TidalBackend-7] mopidy_tidal.library
May 20 11:39:17 raspberrypi mopidy[3163]:   Returning 1 tracks
May 20 11:39:17 raspberrypi mopidy[3163]: INFO     2020-05-20 11:39:17,290 [3163:TidalBackend-7] mopidy_tidal.playback
May 20 11:39:17 raspberrypi mopidy[3163]:   TIDAL uri: tidal:track:4889383:85081120:85081126
May 20 11:39:18 raspberrypi mopidy[3163]: ERROR    2020-05-20 11:39:18,666 [3163:Core-8] mopidy.core.playback
May 20 11:39:18 raspberrypi mopidy[3163]:   TidalBackend backend caused an exception.
May 20 11:39:18 raspberrypi mopidy[3163]: Traceback (most recent call last):
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/mopidy/core/playback.py", line 349, in _change
May 20 11:39:18 raspberrypi mopidy[3163]:     if not backend.playback.change_track(pending_tl_track.track).get():
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/pykka/_threading.py", line 45, in get
May 20 11:39:18 raspberrypi mopidy[3163]:     _compat.reraise(*self._data['exc_info'])
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/pykka/_compat/__init__.py", line 29, in reraise
May 20 11:39:18 raspberrypi mopidy[3163]:     raise value
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/pykka/_actor.py", line 193, in _actor_loop
May 20 11:39:18 raspberrypi mopidy[3163]:     response = self._handle_receive(envelope.message)
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/pykka/_actor.py", line 299, in _handle_receive
May 20 11:39:18 raspberrypi mopidy[3163]:     return callee(*message.args, **message.kwargs)
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/mopidy/backend.py", line 248, in change_track
May 20 11:39:18 raspberrypi mopidy[3163]:     uri = self.translate_uri(track.uri)
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/mopidy_tidal/playback.py", line 16, in translate_uri
May 20 11:39:18 raspberrypi mopidy[3163]:     newurl = self.backend._session.get_media_url(track_id)
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/tidalapi/__init__.py", line 243, in get_media_url
May 20 11:39:18 raspberrypi mopidy[3163]:     r = self.request('GET', 'tracks/%s/streamUrl' % track_id, params)
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/local/lib/python3.7/dist-packages/tidalapi/__init__.py", line 112, in request
May 20 11:39:18 raspberrypi mopidy[3163]:     request.raise_for_status()
May 20 11:39:18 raspberrypi mopidy[3163]:   File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
May 20 11:39:18 raspberrypi mopidy[3163]:     raise HTTPError(http_error_msg, response=self)
May 20 11:39:18 raspberrypi mopidy[3163]: requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.tidalhifi.com/v1/tracks/85081126/streamUrl?sessionId=45057001-46ab-411e-85d
May 20 11:39:18 raspberrypi mopidy[3163]: WARNING  2020-05-20 11:39:18,705 [3163:Core-8] mopidy.core.tracklist
May 20 11:39:18 raspberrypi mopidy[3163]:   Track is not playable: tidal:track:4889383:85081120:85081126

```

**Comment by nuggetman** [comment]
Have started getting the same issue, am also guessing Tidal changed something..

**Comment by tehkillerbee** [comment]
Thanks for the heads-up, I get the same issue. Looks like tidal has been changing things lately. The issue originates from the python-tidal library and has been fixed in the latest version of python-tidal. I updated python-tidal and restarted mopidy - that fixed the issue for me. Make sure you install the latest version available from [python-tidal git repository](https://github.com/tamland/python-tidal).

See the following [issue](https://github.com/tamland/python-tidal/issues/37) for more details.
